### ISSUE TITLE
Safari saves only one of “Set-Cookie” headers

### ISSUE THREAD

**Issue by casaout** [expressjs/csurf#189]
I'm using the node.js server with the csurf library and it's hosted on Microsoft Azure. The csurf library sends a response cookie with a x-csrf-token, but it is not saved in Safari. 

Also in every response Azure sends an ARRAffinity cookie which is properly saved in Safari. 

Is it some bug in Safari/Csurf and only one cookie is saved, or can I change it somehow?

**Comment by dougwilson** [comment]
This module doesn't send two set-cookie headers, so I'm not sure what you're seeing. It can be configured to send a single cookie (named `_csrf` by default) if you've configured it to do so. Is there a way to can provide the following information?

1. Version of this module you're using
2. Version of Node.js you're using
3. A complete app that I can run without modification that demonstrates the issue
4. Instructions for how to reproduce said issue

**Comment by casaout** [comment]
hi @dougwilson Thanks for your response and sorry for my delay with providing more details. Our developer is currently on vacation, and I couldn't finish providing the replication-package. We'll do so as soon as he returns.

**Comment by casaout** [comment]
Apologies for the delay. Our developer just completed the replication package:
* Source Code Server (Github): https://github.com/OkomoHQ/ReproductionPackage_CsrfTokenIssueNodeJS
* Source Code Client (Github): https://github.com/OkomoHQ/ReproductionPackage_CsrfTokenIssueAngular

Details:
* Csurf Version: 1.9.0
* Node.js Version: 0.10.40
* We are hosting it with two Azure App Services.
* Test-site: https://testangularcsrf.azurewebsites.net/home -> click "Test csrf" and the text will update with the status
* The test works well in Chrome and Firefox, but **fails in Safari** (on Mac and iOS) with error 403

Any pointers towards solving the issue would be very much appreciated. thank you!

**Comment by dougwilson** [comment]
Thanks for that. Nothing looks wrong looking at it, and since I don't have access to a Safai browser, I don't think I can help beyond that, sadly.

**Comment by casaout** [comment]
Thank you @dougwilson Do you maybe know someone else (with a Safari browser) who could help us or possibly had a similar issue in the past? 

I still think that this use case cannot be that special and we've done something very simple very wrong..., or it happened to others before...

Any pointers are very welcome!

**Comment by dougwilson** [comment]
My circle like me is pretty much windows and android. I'm hoping that perhaps someone will come by this issue who can help diagnose what is happening based on your example.

**Comment by casaout** [comment]
me too :) 
Could it maybe help if you add other labels, so it's clear the info is there, but we hope for help? Thanks again.

**Comment by dougwilson** [comment]
Done :+1:

**Comment by dougwilson** [comment]
I'm going to close this as it seems no one is interested in helping, and keeping issues open makes people think the module is abandoned, so I'm trying to clean then up.

**Comment by casaout** [comment]
Thanks @dougwilson for remininding me to give an update. We actually removed the csurf library at some point, and still had issues when trying to store cookies in Safari. After a long time googling around and further investigating the issue, we found out that Safari has a stricter cookie policy (compared to the other browsers), in that it only allows to store a cookie if it's sent from the same domain/subdomain.
